pkg/transport/sync-service/sync_service.go

```diff
@@ -148,36 +150,48 @@ func (s *SyncService) handleBundles(ctx context.Context) {
 		case <-ctx.Done():
 			return
 		case objectMetaData := <-s.bundlesMetaDataChan:
-			var buf bytes.Buffer
-			if !s.client.FetchObjectData(objectMetaData, &buf) {
-				s.logError(errSyncServiceReadFailed, "failed to read bundle from sync service", objectMetaData)
-				continue
-			}
-			// sync-service does not need to check for leafHubName since we assume actual selective-distribution.
-			decompressedPayload, err := s.decompressPayload(buf.Bytes(), objectMetaData)
-			if err != nil {
-				s.logError(err, "failed to decompress bundle bytes", objectMetaData)
-				continue
+			if err := s.handleBundle(objectMetaData); err != nil {
+				s.logError(err, "failed to handle bundle", objectMetaData)
 			}
+		}
+	}
+}
 
-			customBundleRegistration, found := s.customBundleIDToRegistrationMap[objectMetaData.ObjectID]
-			if !found { // received generic bundle
-				if err := s.syncGenericBundle(decompressedPayload); err != nil {
-					s.logError(err, "failed to parse bundle", objectMetaData)
-					return
-				}
-			} else {
-				if err := helpers.SyncCustomBundle(customBundleRegistration, decompressedPayload); err != nil {
-					s.logError(err, "failed to parse bundle", objectMetaData)
-					return
-				}
-			}
-			// mark received
-			if err := s.client.MarkObjectReceived(objectMetaData); err != nil {
-				s.logError(err, "failed to report object received to sync service", objectMetaData)
-			}
+func (s *SyncService) handleBundle(objectMetaData *client.ObjectMetaData) error {
+	var buf bytes.Buffer
+	if !s.client.FetchObjectData(objectMetaData, &buf) {
+		return errSyncServiceReadFailed
+	}
+	// sync-service does not need to check for leafHubName since we assume actual selective-distribution.
+	decompressedPayload, err := s.decompressPayload(buf.Bytes(), objectMetaData)
+	if err != nil {
+		return fmt.Errorf("failed to decompress bundle bytes - %w", err)
+	}
+
+	// get msgID
+	msgIDTokens := strings.Split(objectMetaData.ObjectID, ".") // object id is LH_ID.MSG_ID
+	if len(msgIDTokens) != msgIDHeaderTokensLength {
+		return fmt.Errorf("expecting ObjectID of format LH_ID.MSG_ID - %w", errMessageIDWrongFormat)
+	}
+
+	msgID := msgIDTokens[1]
```

Comment:
this change is incorrect.
we add prefix of LH_ID only for messages that are sent to the current LH.
for messages that are broadcasted we don't add it.
see the logic of the HoH here (you wrote it) - 
https://github.com/stolostron/hub-of-hubs-spec-transport-bridge/blob/main/pkg/transport/sync-service/sync_service.go#L124-#L126